commonmark/src/main/java/org/commonmark/content/TextContentRenderer.java

```diff
@@ -0,0 +1,253 @@
+            visitChildren(blockQuote);
+            textContent.write('»');
+
+            writeEndOfLine(blockQuote, null);
+        }
+
+        @Override
+        public void visit(BulletList bulletList) {
+            visitChildren(bulletList);
+            writeEndOfLine(bulletList, null);
+        }
+
+        @Override
+        public void visit(Code code) {
+            textContent.write('\"');
+            textContent.write(code.getLiteral());
+            textContent.write('\"');
+        }
+
+        @Override
+        public void visit(FencedCodeBlock fencedCodeBlock) {
+            if (stripNewlines) {
+                textContent.writeStripped(fencedCodeBlock.getLiteral());
+                writeEndOfLine(fencedCodeBlock, null);
+            } else {
+                textContent.write(fencedCodeBlock.getLiteral());
+            }
+        }
+
+        @Override
+        public void visit(HardLineBreak hardLineBreak) {
+            writeEndOfLine(hardLineBreak, null);
+        }
+
+        @Override
+        public void visit(Heading heading) {
+            visitChildren(heading);
+            writeEndOfLine(heading, ':');
+        }
+
+        @Override
+        public void visit(ThematicBreak thematicBreak) {
+            if (!stripNewlines) {
+                textContent.write("***");
+            }
+            writeEndOfLine(thematicBreak, null);
+        }
+
+        @Override
+        public void visit(HtmlInline htmlInline) {
+            writeText(htmlInline.getLiteral());
+        }
+
+        @Override
+        public void visit(HtmlBlock htmlBlock) {
+            writeText(htmlBlock.getLiteral());
+        }
+
+        @Override
+        public void visit(Image image) {
+            writeLink(image, image.getTitle(), image.getDestination());
+        }
+
+        @Override
+        public void visit(IndentedCodeBlock indentedCodeBlock) {
+            if (stripNewlines) {
+                textContent.writeStripped(indentedCodeBlock.getLiteral());
+                writeEndOfLine(indentedCodeBlock, null);
+            } else {
+                textContent.write(indentedCodeBlock.getLiteral());
+            }
+        }
+
+        @Override
+        public void visit(Link link) {
+            writeLink(link, link.getTitle(), link.getDestination());
+        }
+
+        @Override
+        public void visit(ListItem listItem) {
+            if (orderedListCounter > 0) {
+                textContent.write(String.valueOf(orderedListCounter) + ". ");
+                visitChildren(listItem);
+                writeEndOfLine(listItem, null);
+                orderedListCounter++;
+            } else {
+                if (!stripNewlines) {
+                    textContent.write("- ");
+                }
+                visitChildren(listItem);
+                writeEndOfLine(listItem, null);
+            }
+        }
+
+        @Override
+        public void visit(OrderedList orderedList) {
+            orderedListCounter = 1;
```

Comment:
You're right :) Fixed here https://github.com/atlassian/commonmark-java/pull/58/commits/2d520cc8736c0b0f078f0ffa529dabd479c4ef66